### PR TITLE
Correct null value bug

### PR DESCRIPTION
**Pull request by peternewell**
https://bugs.launchpad.net/or/+bug/1916688

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs [hunk 1]
```diff
@@ -547,8 +547,9 @@ public override void Update(float elapsedClockSeconds)
                     // depending upon whether steam brake fitted or not, calculate brake force to be applied
                     if (LocomotiveSteamBrakeFitted && (Car.WagonType == MSTSWagon.WagonTypes.Engine || Car.WagonType == MSTSWagon.WagonTypes.Tender))
                     {
-                        f = Car.MaxBrakeForceN * Math.Min(SteamBrakeCylinderPressurePSI / lead.MaxBoilerPressurePSI, 1);
-                    }
+                    var leadLocomotiveMaxBoilerPressurePSI = lead == null ? 200.0f : (lead.MaxBoilerPressurePSI);
+                    f = Car.MaxBrakeForceN * Math.Min(SteamBrakeCylinderPressurePSI / leadLocomotiveMaxBoilerPressurePSI, 1);
+                }
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs:552]:
```suggestion
                        var leadLocomotiveMaxBoilerPressurePSI = lead?.MaxBoilerPressurePSI ?? 200.0f;
                        f = Car.MaxBrakeForceN * Math.Min(SteamBrakeCylinderPressurePSI / leadLocomotiveMaxBoilerPressurePSI, 1);
                    }
```

Comment (YoRyan) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs:552]: Match the existing indentation and use the nullable operators.

**Review by YoRyan — CHANGES_REQUESTED** [review]

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs (hunk 1 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs]: Indentations now corrected.

My preference is to leave the null as it is. I want to do this as I think that this is more understandable for "newbie" coders who don't have your skills. In its abbreviated form it doesn't indicate that it is linked to a null. Is there any difference in processing etc to justify the more abbreviated form?

Comment (YoRyan) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs]: No, shouldn't be a performance-related difference.

Nullables are a really handy syntax to have, particularly when you're dealing with chains of properties that may or may not be null, for example:

```C#
return Viewer?.SomeLocomotive?.SomeBoiler?.MaxBoilerPressurePSI ?? 200f;
```

versus

```C#
if (Viewer != null
        && Viewer.SomeLocomotive != null
        && Viewer.SomeLocomotive.SomeBoiler != null
        && Viewer.SomeLocomotive.SomeBoiler.MaxBoilerPressurePSI != null)
    return Viewer.SomeLocomotive.SomeBoiler.MaxBoilerPressurePSI;
else
    return 200f;
```

but the choice is up to you. I certainly won't impose my personal preferences on anyone.

Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs]: In this instance, I prefer to retain my coding for the nulls.

Are you happy to approve on this basis?

Comment (YoRyan) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Brakes/MSTS/VacuumSinglePipe.cs]: Certainly.

**Review by YoRyan — APPROVED** [review]

**Pull request merged**